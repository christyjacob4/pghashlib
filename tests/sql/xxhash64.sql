-- Test basic hash functionality with text
SELECT xxhash64('hello world');

-- Test hash with different text inputs
SELECT xxhash64('test string');
SELECT xxhash64('another test');

-- Test text input with custom seed
SELECT xxhash64('hello world', 42);
SELECT xxhash64('hello world', 123);

-- Test bytea input
SELECT xxhash64('hello world'::bytea);

-- Test bytea input with custom seed
SELECT xxhash64('hello world'::bytea, 42);

-- Test integer input
SELECT xxhash64(12345);
SELECT xxhash64(-12345);

-- Test integer input with custom seed
SELECT xxhash64(12345, 42);
SELECT xxhash64(-12345, 123);

-- Test consistency (same input should give same hash)
SELECT xxhash64('consistent test') = xxhash64('consistent test');

-- Test seed effect (same input, different seeds should give different hashes)
SELECT xxhash64('seed test', 1) != xxhash64('seed test', 2);

-- Test empty string
SELECT xxhash64('');

-- Test single character
SELECT xxhash64('a');

-- Test long string
SELECT xxhash64('Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.');

-- Test function properties
SELECT 
    proname,
    provolatile,
    proisstrict
FROM pg_proc 
WHERE proname = 'xxhash64'
ORDER BY proname, proargtypes;

-- Test extension metadata
SELECT 
    extname,
    extversion
FROM pg_extension 
WHERE extname = 'hashlib';