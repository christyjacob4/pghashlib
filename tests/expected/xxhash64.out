-- Test basic hash functionality with text
SELECT xxhash64('hello world');
      xxhash64       
---------------------
 5020219685658847592
(1 row)

-- Test hash with different text inputs
SELECT xxhash64('test string');
       xxhash64       
----------------------
 -1593496561654517352
(1 row)

SELECT xxhash64('another test');
      xxhash64       
---------------------
 4558306101224602503
(1 row)

-- Test text input with custom seed
SELECT xxhash64('hello world', 42);
      xxhash64       
---------------------
 7620854247404556961
(1 row)

SELECT xxhash64('hello world', 123);
      xxhash64       
---------------------
 9195626791449634285
(1 row)

-- Test bytea input
SELECT xxhash64('hello world'::bytea);
      xxhash64       
---------------------
 5020219685658847592
(1 row)

-- Test bytea input with custom seed
SELECT xxhash64('hello world'::bytea, 42);
      xxhash64       
---------------------
 7620854247404556961
(1 row)

-- Test integer input
SELECT xxhash64(12345);
      xxhash64       
---------------------
 3171224636284295705
(1 row)

SELECT xxhash64(-12345);
      xxhash64       
---------------------
 8529010596457369546
(1 row)

-- Test integer input with custom seed
SELECT xxhash64(12345, 42);
      xxhash64       
---------------------
 -122588266845139798
(1 row)

SELECT xxhash64(-12345, 123);
       xxhash64       
----------------------
 -9205233108327392353
(1 row)

-- Test consistency (same input should give same hash)
SELECT xxhash64('consistent test') = xxhash64('consistent test');
 ?column? 
----------
 t
(1 row)

-- Test seed effect (same input, different seeds should give different hashes)
SELECT xxhash64('seed test', 1) != xxhash64('seed test', 2);
 ?column? 
----------
 t
(1 row)

-- Test empty string
SELECT xxhash64('');
       xxhash64       
----------------------
 -1205034819632174695
(1 row)

-- Test single character
SELECT xxhash64('a');
       xxhash64       
----------------------
 -3292477735350538661
(1 row)

-- Test long string
SELECT xxhash64('Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.');
       xxhash64       
----------------------
 -2121420406496471266
(1 row)

-- Test function properties
SELECT 
    proname,
    provolatile,
    proisstrict
FROM pg_proc 
WHERE proname = 'xxhash64'
ORDER BY proname, proargtypes;
 proname  | provolatile | proisstrict 
----------+-------------+-------------
 xxhash64 | i           | t
 xxhash64 | i           | t
 xxhash64 | i           | t
 xxhash64 | i           | t
 xxhash64 | i           | t
 xxhash64 | i           | t
(6 rows)

-- Test extension metadata
SELECT 
    extname,
    extversion
FROM pg_extension 
WHERE extname = 'hashlib';
 extname | extversion 
---------+------------
 hashlib | 0.0.1
(1 row)

